Content.Server/Antag/AntagRandomObjectivesSystem.cs

```diff
@@ -32,17 +33,25 @@ private void OnAntagSelected(Entity<AntagRandomObjectivesComponent> ent, ref Aft
         }
 
         var difficulty = 0f;
-        foreach (var set in ent.Comp.Sets)
+        var assignedTargetObjectives = new HashSet<(string? title, string? target)>();
+        foreach (var set in ent.Comp.Sets.Where(set => _random.Prob(set.Prob)))
         {
-            if (!_random.Prob(set.Prob))
-                continue;
-
             for (var pick = 0; pick < set.MaxPicks && ent.Comp.MaxDifficulty > difficulty; pick++)
             {
                 var remainingDifficulty = ent.Comp.MaxDifficulty - difficulty;
                 if (_objectives.GetRandomObjective(mindId, mind, set.Groups, remainingDifficulty) is not { } objective)
                     continue;
 
+                if (TryComp<TargetObjectiveComponent>(objective, out var targetObjective))
+                {
+                    var title = targetObjective.Title;
+                    var target = ToPrettyString(targetObjective.Target);
```

Comment:
This method a little weird! Using strings like this is almost always not the correct way of going about it.

I haven't looked at objective code in a while, but I think a better approach might be listening for the `ObjectiveAssignedEvent` in the `TargetObjectiveSystem` and canceling the assignment there if you already have a target targeting the same person as the one that is about to be assigned.

I'm not 100% sure that would work but it would be a cleaner solution if it does!

Reply:
Yeah this is absolutely not the right way to go about this. In fact you literally have an EntityUid, you could store that.